### PR TITLE
Added CreateDirectMessage

### PR DESCRIPTION
**Pull request by facundomedica**

### CODE REVIEW

**Comment by facundomedica** [comment]
@geekgonecrazy I don't know why it shows commits that were already merged in this repo, sorry about that.

File: models/im.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package models
+
+import "time"
+
+type Room struct {
```
Comment (geekgonecrazy) [inline models/im.go]: Room inside of a file named im is kind of confusing. 

Also room and channel are the same thing.  Internally both are rooms.  Not sure if we need this type on this end,

File: models/im.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	UpdatedAt time.Time `json:"_updatedAt"`
+	Type      string    `json:"t"`
+	Messages  int       `json:"msgs"`
+	Timestamp time.Time `json:"ts"`
+	Meta      struct {
+		Revision int   `json:"revision"`
+		Created  int64 `json:"created"`
+		Version  int   `json:"version"`
+	} `json:"meta"`
+	Loki      int      `json:"$loki"`
```
Comment (geekgonecrazy) [inline models/im.go]: I think this should be gone.  It might used to have been used.  But loki internally should be gone.

File: rest/users.go [hunk 3]
```diff
@@ -21,6 +25,31 @@ type logonResponse struct {
 	} `json:"data"`
 }
 
+type CreateUserResponse struct {
+	Status
+	User struct {
```
Comment (geekgonecrazy) [inline rest/users.go]: Is this different then the existing user model?

File: realtime/channels.go [hunk 4]
```diff
@@ -4,7 +4,7 @@ import (
 	"log"
 
 	"github.com/Jeffail/gabs"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/channels.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: models/im.go (hunk 2 above)
Comment (geekgonecrazy) [inline models/im.go]:
```suggestion
```

File: realtime/client_test.go [hunk 5]
```diff
@@ -4,8 +4,8 @@ import (
 	"net/url"
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/common_testing"
```
Comment (geekgonecrazy) [inline realtime/client_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
```

File: realtime/client_test.go [hunk 6]
```diff
@@ -4,8 +4,8 @@ import (
...
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/client_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: realtime/messages.go [hunk 7]
```diff
@@ -7,7 +7,7 @@ import (
 	"time"
 
 	"github.com/Jeffail/gabs"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/messages.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: realtime/messages_test.go [hunk 8]
```diff
@@ -3,7 +3,7 @@ package realtime
 import (
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/messages_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: realtime/permissions.go [hunk 9]
```diff
@@ -4,7 +4,7 @@ import (
 	"log"
 
 	"github.com/Jeffail/gabs"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/permissions.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: realtime/settings.go [hunk 10]
```diff
@@ -4,7 +4,7 @@ import (
 	"log"
 
 	"github.com/Jeffail/gabs"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/settings.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: realtime/users.go [hunk 11]
```diff
@@ -6,7 +6,7 @@ import (
 	"strconv"
 
 	"github.com/Jeffail/gabs"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline realtime/users.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/channels.go [hunk 12]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"net/url"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/channels.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/channels_test.go [hunk 13]
```diff
@@ -3,7 +3,7 @@ package rest
 import (
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/channels_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/client_test.go [hunk 14]
```diff
@@ -4,9 +4,9 @@ import (
 	"net/url"
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/realtime"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/common_testing"
```
Comment (geekgonecrazy) [inline rest/client_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
```

File: rest/client_test.go [hunk 15]
```diff
@@ -4,9 +4,9 @@ import (
...
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/client_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/client_test.go [hunk 16]
```diff
@@ -4,9 +4,9 @@ import (
...
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/realtime"
```
Comment (geekgonecrazy) [inline rest/client_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/realtime"
```

File: rest/im.go [hunk 17]
```diff
@@ -0,0 +1,31 @@
+package rest
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/im.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/information.go [hunk 18]
```diff
@@ -3,7 +3,7 @@ package rest
 import (
 	"net/url"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/information.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/information_test.go [hunk 19]
```diff
@@ -4,7 +4,7 @@ import (
 	"net/url"
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/common_testing"
```
Comment (geekgonecrazy) [inline rest/information_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
```

File: rest/messages.go [hunk 20]
```diff
@@ -8,7 +8,7 @@ import (
 	"net/url"
 	"strconv"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/messages.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/messages_test.go [hunk 21]
```diff
@@ -3,7 +3,7 @@ package rest
 import (
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/messages_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/users.go [hunk 22]
```diff
@@ -7,7 +7,7 @@ import (
 	"net/url"
 	"time"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/models"
```
Comment (geekgonecrazy) [inline rest/users.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
```

File: rest/users_test.go [hunk 23]
```diff
@@ -3,7 +3,7 @@ package rest
 import (
 	"testing"
 
-	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
+	"github.com/facundomedica/Rocket.Chat.Go.SDK/common_testing"
```
Comment (geekgonecrazy) [inline rest/users_test.go]:
```suggestion
	"github.com/RocketChat/Rocket.Chat.Go.SDK/common_testing"
```

File: models/im.go (hunk 1 above)
Comment (geekgonecrazy) [reply models/im.go]: will just change this later maybe.

**Comment by geekgonecrazy** [comment]
@facundomedica if you can make these changes i'll get this merged

**Comment by facundomedica** [comment]
@geekgonecrazy we've been using my fork at work quite extensively and we've added some changes that maybe shouldn't be on the RockeChat's official SDK. Feel free to close this PR until I get the time to refactor.

**Comment by geekgonecrazy** [comment]
We can definitely close if you want to instead open a new PR later on.  I understand not wanting to disrupt your work

**Pull request closed without merging**